site/src/pages/api/blocks/create.api.ts

```diff
@@ -26,20 +37,61 @@ export default createAuthenticatedHandler<
     bodyValidator("npmPackageName").isString().notEmpty(),
   )
   .post(async (req, res) => {
+    if (!shouldAllowBlockPublishing) {
+      return res
+        .status(501)
+        .json(formatErrors({ msg: "Publishing is not supported." }));
+    }
+
     const errors = validationResult(req);
     if (!errors.isEmpty()) {
       return res.status(400).json(formatErrors(...errors.array()));
     }
 
-    const { db, user } = req;
-    const { blockName, npmPackageName } = req.body;
+    const {
+      db,
+      user: { shortname },
+    } = req;
+
+    if (!shortname) {
+      return res.status(403).json(
+        formatErrors({
+          code: "SIGNUP_INCOMPLETE",
+          msg: "You must be signed up to complete this request",
+        }),
+      );
+    }
+
+    const { blockName: untransformedBlockName, npmPackageName } = req.body;
+
+    const slugifiedBlockName = generateSlug(untransformedBlockName);
```

Comment:
`blockName` is required to be specified and is supposed to be a slug, this is just making sure it is.

I've updated it to throw an error if it isn't instead of silently changing it, in https://github.com/blockprotocol/blockprotocol/pull/548/commits/5baef1df09ca34a3db9a18e381e29a0e0ae88600